test/abilities/mycelium_might.test.ts

```diff
@@ -28,80 +27,59 @@ describe("Abilities - Mycelium Might", () => {
     game.override.disableCrits();
     game.override.enemySpecies(Species.SHUCKLE);
     game.override.enemyAbility(Abilities.CLEAR_BODY);
-    game.override.enemyMoveset([MoveId.QUICK_ATTACK, MoveId.QUICK_ATTACK, MoveId.QUICK_ATTACK, MoveId.QUICK_ATTACK]);
+    game.override.enemyMoveset(MoveId.QUICK_ATTACK);
     game.override.ability(Abilities.MYCELIUM_MIGHT);
     game.override.moveset([MoveId.QUICK_ATTACK, MoveId.BABY_DOLL_EYES]);
   });
 
+  const getTurnOrder = () => {
+    return game.scene
+      .getField(true)
+      .sort((a, b) => a.turnData.order - b.turnData.order)
+      .map((p) => p.getBattlerIndex());
+  };
+
   /**
    * References:
    * https://bulbapedia.bulbagarden.net/wiki/Mycelium_Might_(Ability)
    * https://bulbapedia.bulbagarden.net/wiki/Priority
    * https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/page-24
    **/
 
-  it("will move last in its priority bracket and ignore protective abilities", async () => {
-    await game.startBattle([Species.REGIELEKI]);
+  it("should make the source move last in its priority bracket and ignore protective abilities when using a status move", async () => {
+    await game.classicMode.startBattle([Species.REGIELEKI]);
 
     const enemyPokemon = game.scene.getEnemyPokemon();
-    const playerIndex = game.scene.getPlayerPokemon()?.getBattlerIndex();
-    const enemyIndex = enemyPokemon?.getBattlerIndex();
 
     game.move.select(MoveId.BABY_DOLL_EYES);
 
-    await game.phaseInterceptor.to(TurnStartPhase, false);
-    const phase = game.scene.getCurrentPhase() as TurnStartPhase;
-    const speedOrder = phase.getSpeedOrder();
-    const commandOrder = phase.getCommandOrder();
-    // The opponent Pokemon (without Mycelium Might) goes first despite having lower speed than the player Pokemon.
-    // The player Pokemon (with Mycelium Might) goes last despite having higher speed than the opponent.
-    expect(speedOrder).toEqual([playerIndex, enemyIndex]);
-    expect(commandOrder).toEqual([enemyIndex, playerIndex]);
-    await game.phaseInterceptor.to(TurnEndPhase);
-
-    // Despite the opponent's ability (Clear Body), its ATK stat stage is still reduced.
+    await game.phaseInterceptor.to("BerryPhase", false);
+
+    expect(getTurnOrder()).toEqual([BattlerIndex.ENEMY, BattlerIndex.PLAYER]);
     expect(enemyPokemon?.getStatStage(Stat.ATK)).toBe(-1);
   }, 20000);
```

Comment:
```suggestion
  },
);
```
since the test is being "modernized" a little already